### PR TITLE
add scsv and pickle files processing scripts

### PR DESCRIPTION
**Pull request by acruikshank-wm**
## Problem

While working on the VectorSearchChain I created some scripts to handle the [file format] -> json -> embeddings -> pinecone workflow. I neglected to check in the scripts that handle csv -> json and pickel -> json conversion. These scripts are meant for local data processing and should not be considered production ready.

## Solution

1. Add missing scripts

### CODE REVIEW

File: scripts/csv_to_json.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import sys
+import json
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter)
+# parser.add_argument("-f", "--file", help="csv file to read from")
+
+args = vars(parser.parse_args())
+
+def main(args):
```
Comment (wmjrosenblatt) [inline scripts/csv_to_json.py:11]: The args are not used in this method

File: scripts/csv_to_json.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  data = csv.DictReader(sys.stdin)
```
Comment (wmjrosenblatt) [inline scripts/csv_to_json.py:12]: Do we wanna do any error checking for poorly formatted documents?

File: scripts/pickle_to_json.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import pandas as pd
+import pinecone
```
Comment (wmjrosenblatt) [inline scripts/pickle_to_json.py:2]: We could probably just import init from pinecone

File: scripts/pickle_to_json.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import uuid
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter)
+parser.add_argument("-k", "--api_key", type=str, help="API Key for Pinecone")
+parser.add_argument("-emb", "--embedding_file", help="File location for embeddings (must be .pkl file)")
+
+args = vars(parser.parse_args())
+
+def createUUID():
```
Comment (wmjrosenblatt) [inline scripts/pickle_to_json.py:12]: This method seems to be unused

File: scripts/pickle_to_json.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    return str(uuid.uuid4())
+
+## Main Function ##
+def main(args):
+    pinecone.init(api_key=args["api_key"], environment="northamerica-northeast1-gcp")
+    embedding_filepath = args["embedding_file"]
+
+    df_emb = pd.read_pickle(embedding_filepath)
```
Comment (wmjrosenblatt) [inline scripts/pickle_to_json.py:20]: Do we wanna do any error checking for poorly formatted files?

File: scripts/csv_to_json.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import csv
+import sys
+import json
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+parser = ArgumentParser(formatter_class=ArgumentDefaultsHelpFormatter)
+# parser.add_argument("-f", "--file", help="csv file to read from")
```
[the remark this answers is not in the document]
Comment (acruikshank-wm) [reply scripts/csv_to_json.py]: good catch

File: scripts/csv_to_json.py (hunk 2 above)
Comment (acruikshank-wm) [reply scripts/csv_to_json.py:14]: These scripts are not really for use for general users. Error checking CSVs is more effort than this script is worth. I added some documentation on the expected format.

File: scripts/pickle_to_json.py (hunk 3 above)
Comment (acruikshank-wm) [reply scripts/pickle_to_json.py]: Good catch. We actually don't need Pinecone at all in this script. It was leftover cruft.

File: scripts/pickle_to_json.py (hunk 5 above)
Comment (acruikshank-wm) [reply scripts/pickle_to_json.py:14]: This script actually converts any Pkl file to JSON so there aren't really any failure modes beyond a general failure to parse. It'll error out if parsing fails.

**Pull request merged**